code/modules/reagents/reactions/reaction_recipe_food.dm

```diff
@@ -4,29 +4,78 @@
 	abstract_type = /decl/chemical_reaction/recipe/food
 	var/obj_result
 
-/decl/chemical_reaction/recipe/food/on_reaction(var/datum/reagents/holder, var/created_volume, var/reaction_flags)
+/decl/chemical_reaction/recipe/food/on_reaction(datum/reagents/holder, created_volume, reaction_flags, list/reaction_data)
 	..()
 	var/location = get_turf(holder.get_reaction_loc(chemical_reaction_flags))
 	if(obj_result && isturf(location))
 		for(var/i = 1, i <= created_volume, i++)
-			create_food(location)
+			create_food(location, reaction_data)
 
-/decl/chemical_reaction/recipe/food/proc/create_food(location)
+/decl/chemical_reaction/recipe/food/proc/create_food(location, list/data)
 	var/atom/movable/food = new obj_result
 	if(istype(food))
 		food.dropInto(location)
+	if(length(data) && istype(food, /obj/item/food))
+		var/obj/item/food/food_item = food
+		food_item.set_nutriment_data(data)
 	return food
 
-/decl/chemical_reaction/recipe/food/cheesewheel
+/decl/chemical_reaction/recipe/food/dairy
+	abstract_type = /decl/chemical_reaction/recipe/food/dairy
+	var/static/list/milk_data_keys = list(
+		"milk_donor",
+		"milk_color",
+		"milk_name",
+		"cheese_name",
+		"cheese_color"
+	)
+
+/decl/chemical_reaction/recipe/food/dairy/send_data(var/datum/reagents/holder, var/reaction_limit)
+	. = ..()
+	for(var/reagent in required_reagents)
+		var/list/data = LAZYACCESS(holder.reagent_data, reagent)
+		if(!islist(data))
+			continue
+		for(var/milk_key in milk_data_keys)
+			var/milk_val = data[milk_key]
+			if(isnull(milk_val))
+				continue
+			LAZYSET(., milk_key, milk_val)
```

Comment:
this feels like mixing 99u of cow's milk and 1u of goat's milk would produce feta as long as the goat's milk was added second. hm. not sure if there's a good way to fix that or if there's even a point in bothering to try